### ISSUE TITLE
[Bug] The MAE training loss becomes nan after one iteration

### ISSUE THREAD

**Issue by windygoo** [open-mmlab/mmpretrain#1632]
### Branch

main branch (mmpretrain version)

### Describe the bug

I use the mae_vit-base-p16_8xb512-amp-coslr-1600e_in1k.py config file and just change the dataset path to mine, which comprises 1.5M images of size 512 x512. The training loss decreases normally at first (from 0.7+ at epoch-0 to 0.4+ at epoch-147) while suddenly increases from 0.4669 to 0.5318 in the 163-th epoch and then remains as nan.

### Environment

{'sys.platform': 'linux',
 'Python': Python 3.7.15 (default, Oct 12 2022, 19:14:55) [GCC 7.5.0] on linux,
 'CUDA available': True,
 'GPU': 'A100 80G',
 'PyTorch': 'cu111+1.10.0',
 'TorchVision': '0.11.0+cu111',
 'MMEngine': '0.7.3',
 'MMCV': '2.0.0',
 'MMPreTrain': '1.0.0rc8'}

### Other information

2023/06/09 21:58:41 - mmengine - INFO - Epoch(train) [147][300/351] lr: 2.3723e-03 eta: 5 days, 23:52:36 time: 1.0105 data_time: 0.0024 memory: 48596 loss: 0.4665
2023/06/09 21:59:31 - mmengine - INFO - Exp name: mae_vit-base-p16_8xb512-amp-coslr-1600e_pathology_20230609_120904
2023/06/09 22:01:21 - mmengine - INFO - Epoch(train) [148][100/351] lr: 2.3721e-03 eta: 5 days, 23:51:50 time: 1.0365 data_time: 0.0310 memory: 48596 loss: 0.4702
2023/06/09 22:03:01 - mmengine - INFO - Epoch(train) [148][200/351] lr: 2.3720e-03 eta: 5 days, 23:49:46 time: 0.9424 data_time: 0.0308 memory: 48596 loss: 0.4639
2023/06/09 22:04:50 - mmengine - INFO - Epoch(train) [148][300/351] lr: 2.3718e-03 eta: 5 days, 23:49:40 time: 0.9822 data_time: 0.0020 memory: 48596 loss: 0.4674
2023/06/09 22:05:41 - mmengine - INFO - Exp name: mae_vit-base-p16_8xb512-amp-coslr-1600e_pathology_20230609_120904
2023/06/09 22:06:41 - mmengine - INFO - Exp name: mae_vit-base-p16_8xb512-amp-coslr-1600e_pathology_20230609_120904
2023/06/09 22:07:31 - mmengine - INFO - Epoch(train) [149][100/351] lr: 2.3716e-03 eta: 5 days, 23:48:57 time: 0.9751 data_time: 0.0023 memory: 48596 loss: 0.4692
2023/06/09 22:09:14 - mmengine - INFO - Epoch(train) [149][200/351] lr: 2.3714e-03 eta: 5 days, 23:47:35 time: 1.0154 data_time: 0.0023 memory: 48596 loss: 0.4705
2023/06/09 22:10:59 - mmengine - INFO - Epoch(train) [149][300/351] lr: 2.3713e-03 eta: 5 days, 23:46:47 time: 1.0436 data_time: 0.0022 memory: 48596 loss: 0.4674
2023/06/09 22:11:50 - mmengine - INFO - Exp name: mae_vit-base-p16_8xb512-amp-coslr-1600e_pathology_20230609_120904
2023/06/09 22:13:39 - mmengine - INFO - Epoch(train) [150][100/351] lr: 2.3711e-03 eta: 5 days, 23:45:42 time: 0.9467 data_time: 0.0273 memory: 48596 loss: 0.4680
2023/06/09 22:15:17 - mmengine - INFO - Epoch(train) [150][200/351] lr: 2.3709e-03 eta: 5 days, 23:43:13 time: 0.9604 data_time: 0.0022 memory: 48596 loss: 0.4644
2023/06/09 22:17:03 - mmengine - INFO - Epoch(train) [150][300/351] lr: 2.3708e-03 eta: 5 days, 23:42:36 time: 0.9645 data_time: 0.0022 memory: 48596 loss: 0.4710
2023/06/09 22:17:53 - mmengine - INFO - Exp name: mae_vit-base-p16_8xb512-amp-coslr-1600e_pathology_20230609_120904
2023/06/09 22:19:41 - mmengine - INFO - Epoch(train) [151][100/351] lr: 2.3705e-03 eta: 5 days, 23:40:59 time: 1.0113 data_time: 0.0028 memory: 48596 loss: 0.4679
2023/06/09 22:21:25 - mmengine - INFO - Epoch(train) [151][200/351] lr: 2.3704e-03 eta: 5 days, 23:40:00 time: 0.9601 data_time: 0.0021 memory: 48596 loss: 0.4666
2023/06/09 22:23:11 - mmengine - INFO - Epoch(train) [151][300/351] lr: 2.3702e-03 eta: 5 days, 23:39:18 time: 0.9811 data_time: 0.0020 memory: 48596 loss: 0.4719
2023/06/09 22:24:02 - mmengine - INFO - Exp name: mae_vit-base-p16_8xb512-amp-coslr-1600e_pathology_20230609_120904
2023/06/09 22:24:02 - mmengine - INFO - Exp name: mae_vit-base-p16_8xb512-amp-coslr-1600e_pathology_20230609_120904
2023/06/09 22:25:48 - mmengine - INFO - Epoch(train) [152][100/351] lr: 2.3700e-03 eta: 5 days, 23:37:34 time: 1.0612 data_time: 0.1571 memory: 48596 loss: 0.4680
2023/06/09 22:27:34 - mmengine - INFO - Epoch(train) [152][200/351] lr: 2.3698e-03 eta: 5 days, 23:36:43 time: 0.9813 data_time: 0.0031 memory: 48596 loss: 0.4686
2023/06/09 22:29:17 - mmengine - INFO - Epoch(train) [152][300/351] lr: 2.3697e-03 eta: 5 days, 23:35:17 time: 0.9477 data_time: 0.0021 memory: 48596 loss: 0.4686
2023/06/09 22:30:06 - mmengine - INFO - Exp name: mae_vit-base-p16_8xb512-amp-coslr-1600e_pathology_20230609_120904
2023/06/09 22:31:53 - mmengine - INFO - Epoch(train) [153][100/351] lr: 2.3695e-03 eta: 5 days, 23:33:18 time: 0.9408 data_time: 0.0718 memory: 48596 loss: 0.4661
2023/06/09 22:33:36 - mmengine - INFO - Epoch(train) [153][200/351] lr: 2.3693e-03 eta: 5 days, 23:32:00 time: 0.9390 data_time: 0.0949 memory: 48596 loss: 0.4660
2023/06/09 22:35:21 - mmengine - INFO - Epoch(train) [153][300/351] lr: 2.3691e-03 eta: 5 days, 23:31:04 time: 0.9923 data_time: 0.0390 memory: 48596 loss: 0.4685
2023/06/09 22:36:13 - mmengine - INFO - Exp name: mae_vit-base-p16_8xb512-amp-coslr-1600e_pathology_20230609_120904
2023/06/09 22:38:00 - mmengine - INFO - Epoch(train) [154][100/351] lr: 2.3689e-03 eta: 5 days, 23:29:46 time: 0.9602 data_time: 0.0360 memory: 48596 loss: 0.4720
2023/06/09 22:39:42 - mmengine - INFO - Epoch(train) [154][200/351] lr: 2.3688e-03 eta: 5 days, 23:28:09 time: 1.0146 data_time: 0.0022 memory: 48596 loss: 0.4670
2023/06/09 22:41:28 - mmengine - INFO - Exp name: mae_vit-base-p16_8xb512-amp-coslr-1600e_pathology_20230609_120904
2023/06/09 22:41:30 - mmengine - INFO - Epoch(train) [154][300/351] lr: 2.3686e-03 eta: 5 days, 23:27:51 time: 1.0459 data_time: 0.0292 memory: 48596 loss: 0.4693
2023/06/09 22:42:22 - mmengine - INFO - Exp name: mae_vit-base-p16_8xb512-amp-coslr-1600e_pathology_20230609_120904
2023/06/09 22:44:11 - mmengine - INFO - Epoch(train) [155][100/351] lr: 2.3684e-03 eta: 5 days, 23:26:54 time: 0.9603 data_time: 0.0415 memory: 48596 loss: 0.4660
2023/06/09 22:45:52 - mmengine - INFO - Epoch(train) [155][200/351] lr: 2.3682e-03 eta: 5 days, 23:25:04 time: 0.9997 data_time: 0.0021 memory: 48596 loss: 0.4661
2023/06/09 22:47:32 - mmengine - INFO - Epoch(train) [155][300/351] lr: 2.3680e-03 eta: 5 days, 23:23:02 time: 0.9527 data_time: 0.0024 memory: 48596 loss: 0.4693
2023/06/09 22:48:24 - mmengine - INFO - Exp name: mae_vit-base-p16_8xb512-amp-coslr-1600e_pathology_20230609_120904
2023/06/09 22:50:09 - mmengine - INFO - Epoch(train) [156][100/351] lr: 2.3678e-03 eta: 5 days, 23:21:06 time: 0.9207 data_time: 0.0501 memory: 48596 loss: 0.4668
2023/06/09 22:51:47 - mmengine - INFO - Epoch(train) [156][200/351] lr: 2.3676e-03 eta: 5 days, 23:18:38 time: 0.9217 data_time: 0.0026 memory: 48596 loss: 0.4670
2023/06/09 22:53:31 - mmengine - INFO - Epoch(train) [156][300/351] lr: 2.3675e-03 eta: 5 days, 23:17:30 time: 1.0139 data_time: 0.0039 memory: 48596 loss: 0.4647
2023/06/09 22:54:23 - mmengine - INFO - Exp name: mae_vit-base-p16_8xb512-amp-coslr-1600e_pathology_20230609_120904
2023/06/09 22:56:07 - mmengine - INFO - Epoch(train) [157][100/351] lr: 2.3672e-03 eta: 5 days, 23:15:17 time: 0.9711 data_time: 0.1006 memory: 48596 loss: 0.4667
2023/06/09 22:57:47 - mmengine - INFO - Epoch(train) [157][200/351] lr: 2.3671e-03 eta: 5 days, 23:13:18 time: 0.9670 data_time: 0.0259 memory: 48596 loss: 0.4695
2023/06/09 22:58:32 - mmengine - INFO - Exp name: mae_vit-base-p16_8xb512-amp-coslr-1600e_pathology_20230609_120904
2023/06/09 22:59:28 - mmengine - INFO - Epoch(train) [157][300/351] lr: 2.3669e-03 eta: 5 days, 23:11:30 time: 0.9783 data_time: 0.0042 memory: 48596 loss: 0.4676
2023/06/09 23:00:20 - mmengine - INFO - Exp name: mae_vit-base-p16_8xb512-amp-coslr-1600e_pathology_20230609_120904
2023/06/09 23:02:07 - mmengine - INFO - Epoch(train) [158][100/351] lr: 2.3667e-03 eta: 5 days, 23:09:58 time: 0.9937 data_time: 0.0021 memory: 48596 loss: 0.4681
2023/06/09 23:03:46 - mmengine - INFO - Epoch(train) [158][200/351] lr: 2.3665e-03 eta: 5 days, 23:07:37 time: 0.9636 data_time: 0.0021 memory: 48596 loss: 0.4667
2023/06/09 23:05:31 - mmengine - INFO - Epoch(train) [158][300/351] lr: 2.3664e-03 eta: 5 days, 23:06:43 time: 0.9710 data_time: 0.0024 memory: 48596 loss: 0.4738
2023/06/09 23:06:20 - mmengine - INFO - Exp name: mae_vit-base-p16_8xb512-amp-coslr-1600e_pathology_20230609_120904
2023/06/09 23:08:07 - mmengine - INFO - Epoch(train) [159][100/351] lr: 2.3661e-03 eta: 5 days, 23:04:38 time: 0.9720 data_time: 0.0677 memory: 48596 loss: 0.4685
2023/06/09 23:09:49 - mmengine - INFO - Epoch(train) [159][200/351] lr: 2.3660e-03 eta: 5 days, 23:02:55 time: 0.9829 data_time: 0.0023 memory: 48596 loss: 0.4677
2023/06/09 23:11:32 - mmengine - INFO - Epoch(train) [159][300/351] lr: 2.3658e-03 eta: 5 days, 23:01:35 time: 0.9585 data_time: 0.0026 memory: 48596 loss: 0.4663
2023/06/09 23:12:24 - mmengine - INFO - Exp name: mae_vit-base-p16_8xb512-amp-coslr-1600e_pathology_20230609_120904
2023/06/09 23:14:12 - mmengine - INFO - Epoch(train) [160][100/351] lr: 2.3655e-03 eta: 5 days, 23:00:15 time: 0.9940 data_time: 0.0022 memory: 48596 loss: 0.4690
2023/06/09 23:15:42 - mmengine - INFO - Exp name: mae_vit-base-p16_8xb512-amp-coslr-1600e_pathology_20230609_120904
2023/06/09 23:15:51 - mmengine - INFO - Epoch(train) [160][200/351] lr: 2.3654e-03 eta: 5 days, 22:57:58 time: 0.9811 data_time: 0.0027 memory: 48596 loss: 0.4674
2023/06/09 23:17:32 - mmengine - INFO - Epoch(train) [160][300/351] lr: 2.3652e-03 eta: 5 days, 22:56:14 time: 0.9050 data_time: 0.0024 memory: 48596 loss: 0.4675
2023/06/09 23:18:22 - mmengine - INFO - Exp name: mae_vit-base-p16_8xb512-amp-coslr-1600e_pathology_20230609_120904
2023/06/09 23:20:06 - mmengine - INFO - Epoch(train) [161][100/351] lr: 2.3650e-03 eta: 5 days, 22:53:34 time: 0.9724 data_time: 0.1021 memory: 48596 loss: 0.4666
2023/06/09 23:21:47 - mmengine - INFO - Epoch(train) [161][200/351] lr: 2.3648e-03 eta: 5 days, 22:51:41 time: 0.9849 data_time: 0.0530 memory: 48596 loss: 0.4667
2023/06/09 23:23:28 - mmengine - INFO - Epoch(train) [161][300/351] lr: 2.3646e-03 eta: 5 days, 22:49:55 time: 0.9463 data_time: 0.0022 memory: 48596 loss: 0.4710
2023/06/09 23:24:21 - mmengine - INFO - Exp name: mae_vit-base-p16_8xb512-amp-coslr-1600e_pathology_20230609_120904
2023/06/09 23:26:08 - mmengine - INFO - Epoch(train) [162][100/351] lr: 2.3644e-03 eta: 5 days, 22:48:37 time: 0.9541 data_time: 0.0021 memory: 48596 loss: 0.4700
2023/06/09 23:27:46 - mmengine - INFO - Epoch(train) [162][200/351] lr: 2.3642e-03 eta: 5 days, 22:46:16 time: 0.9653 data_time: 0.0022 memory: 48596 loss: 0.4677
2023/06/09 23:29:32 - mmengine - INFO - Epoch(train) [162][300/351] lr: 2.3641e-03 eta: 5 days, 22:45:24 time: 1.0256 data_time: 0.0021 memory: 48596 loss: 0.4652
2023/06/09 23:30:22 - mmengine - INFO - Exp name: mae_vit-base-p16_8xb512-amp-coslr-1600e_pathology_20230609_120904
2023/06/09 23:32:07 - mmengine - INFO - Epoch(train) [163][100/351] lr: 2.3638e-03 eta: 5 days, 22:43:09 time: 0.9675 data_time: 0.1117 memory: 48596 loss: 0.4669
2023/06/09 23:32:50 - mmengine - INFO - Exp name: mae_vit-base-p16_8xb512-amp-coslr-1600e_pathology_20230609_120904
2023/06/09 23:33:51 - mmengine - INFO - Epoch(train) [163][200/351] lr: 2.3636e-03 eta: 5 days, 22:41:48 time: 0.9305 data_time: 0.0023 memory: 48596 loss: 0.5318
2023/06/09 23:35:31 - mmengine - INFO - Epoch(train) [163][300/351] lr: 2.3635e-03 eta: 5 days, 22:39:45 time: 0.9373 data_time: 0.0025 memory: 48596 loss: nan
2023/06/09 23:36:22 - mmengine - INFO - Exp name: mae_vit-base-p16_8xb512-amp-coslr-1600e_pathology_20230609_120904
2023/06/09 23:38:07 - mmengine - INFO - Epoch(train) [164][100/351] lr: 2.3632e-03 eta: 5 days, 22:37:37 time: 0.9780 data_time: 0.0022 memory: 48596 loss: nan
2023/06/09 23:39:45 - mmengine - INFO - Epoch(train) [164][200/351] lr: 2.3630e-03 eta: 5 days, 22:35:12 time: 0.8733 data_time: 0.0027 memory: 48596 loss: nan
2023/06/09 23:41:28 - mmengine - INFO - Epoch(train) [164][300/351] lr: 2.3629e-03 eta: 5 days, 22:33:41 time: 0.9546 data_time: 0.0022 memory: 48596 loss: nan
2023/06/09 23:42:17 - mmengine - INFO - Exp name: mae_vit-base-p16_8xb512-amp-coslr-1600e_pathology_20230609_120904
2023/06/09 23:44:01 - mmengine - INFO - Epoch(train) [165][100/351] lr: 2.3626e-03 eta: 5 days, 22:31:00 time: 0.9755 data_time: 0.1014 memory: 48596 loss: nan
2023/06/09 23:45:41 - mmengine - INFO - Epoch(train) [165][200/351] lr: 2.3624e-03 eta: 5 days, 22:29:00 time: 0.9610 data_time: 0.0021 memory: 48596 loss: nan
2023/06/09 23:47:22 - mmengine - INFO - Epoch(train) [165][300/351] lr: 2.3623e-03 eta: 5 days, 22:27:01 time: 0.9934 data_time: 0.0023 memory: 48596 loss: nan
2023/06/09 23:48:12 - mmengine - INFO - Exp name: mae_vit-base-p16_8xb512-amp-coslr-1600e_pathology_20230609_120904
2023/06/09 23:49:41 - mmengine - INFO - Exp name: mae_vit-base-p16_8xb512-amp-coslr-1600e_pathology_20230609_120904
2023/06/09 23:49:56 - mmengine - INFO - Epoch(train) [166][100/351] lr: 2.3620e-03 eta: 5 days, 22:24:33 time: 0.9772 data_time: 0.1022 memory: 48596 loss: nan
2023/06/09 23:51:33 - mmengine - INFO - Epoch(train) [166][200/351] lr: 2.3618e-03 eta: 5 days, 22:21:53 time: 0.9591 data_time: 0.0681 memory: 48596 loss: nan
2023/06/09 23:53:17 - mmengine - INFO - Epoch(train) [166][300/351] lr: 2.3617e-03 eta: 5 days, 22:20:36 time: 0.9553 data_time: 0.0024 memory: 48596 loss: nan

**Comment by windygoo** [comment]
grad clip solves it

**Comment by windygoo** [comment]
disable AMP solves it

**Comment by waizuichougou** [comment]
I have the same problem. Can you explain the solution in more detail?
